### ISSUE TITLE
if...else in TorchScript  

### ISSUE THREAD

**Issue by xiaodao2049** [pytorch/pytorch#31028]
I found a strange phenomenon when I use TorchScript to train my net. I implement a BatchNorm with mask, and there is a if ... else ... branch in my forward function,  during training, CPU memory is keeping going up slowly and finally lead to CPU OOM. When I remove the if...else... , CPU memory will not go up after reach a certain point.

cc @suo

**Comment by suo** [comment]
Hi, this is definitely not expected behavior. Can you provide us with your environment, what version of PyTorch you are running, and a script that we can run to reproduce the issue? That would help us debug this issue. Thanks!

**Comment by xiaodao2049** [comment]
**Environment**
PyTorch version: 1.3.0+cu100
OS: CentOS linux 7.2.1511
Python version: 3.6

CUDA version:10.1
Driver Version: 418.15
GPU : Tesla P40*4
cuDNN version: cuda-10.0-cudnn-7.3.0

**Comment by xiaodao2049** [comment]
**Environment**
PyTorch version: 1.3.0+cu100
OS: CentOS linux 7.2.1511
Python version: 3.6

CUDA version:10.1
Driver Version: 418.15
GPU : Tesla P40*4
cuDNN version: cuda-10.0-cudnn-7.3.0

@suo Hi, this is my environmen information.
